metricbeat/module/kubernetes/event/event.go

```diff
@@ -107,21 +109,26 @@ func (m *MetricSet) Run(reporter mb.PushReporter) {
 		// skip events happened before watch
 		FilterFunc: func(obj interface{}) bool {
 			eve := obj.(*kubernetes.Event)
-			if kubernetes.Time(&eve.LastTimestamp).Before(now) {
-				return false
-			}
-			return true
+			return !kubernetes.Time(&eve.LastTimestamp).Before(now)
 		},
 		Handler: handler,
 	})
 	// start event watcher
-	m.watcher.Start()
+	err := m.watcher.Start()
+	if err != nil {
+		logp.Debug("kubernetes", "Event watcher hasn't started correctly: %s", err.Error())
+	}
 	<-reporter.Done()
 	m.watcher.Stop()
-	return
 }
 
-func generateMapStrFromEvent(eve *kubernetes.Event, dedotConfig dedotConfig) common.MapStr {
+func (m *MetricSet) reportEvent(obj interface{}, reporter mb.PushReporterV2) {
+	mapStrEvent := generateMapStrFromEvent(obj.(*kubernetes.Event), m.dedotConfig, m.Logger())
+	event := mb.TransformMapStrToEvent("kubernetes", mapStrEvent, nil)
```

Comment:
@tetianakravchenko 
I removed this bit from `main` since it doesn't have a clusterMeta field

```go
	if m.clusterMeta != nil {
		event.RootFields.DeepUpdate(m.clusterMeta)
	}
```

Reply:
do we actually need it? this change seems to be coming from https://github.com/elastic/beats/pull/31341 - I think it shouldn't be a part of backport PR